tests/by-util/test_hashsum.rs

```diff
@@ -480,3 +498,115 @@ fn test_check_with_escape_filename() {
         .succeeds();
     result.stdout_is("\\a\\nb: OK\n");
 }
+
+#[test]
+fn test_check_strict_error() {
+    let scene = TestScenario::new(util_name!());
+    let at = &scene.fixtures;
+
+    at.write("f", "");
+    at.write(
+        "in.md5",
+        "ERR\nERR\nd41d8cd98f00b204e9800998ecf8427e  f\nERR\n",
+    );
+    scene
+        .ccmd("md5sum")
+        .arg("--check")
+        .arg("--strict")
+        .arg(at.subdir.join("in.md5"))
+        .fails()
+        .stderr_contains("WARNING: 3 lines are improperly formatted");
+}
+
+#[test]
+fn test_check_warn() {
+    let scene = TestScenario::new(util_name!());
+    let at = &scene.fixtures;
+
+    at.write("f", "");
+    at.write(
+        "in.md5",
+        "d41d8cd98f00b204e9800998ecf8427e  f\nd41d8cd98f00b204e9800998ecf8427e  f\ninvalid\n",
+    );
+    scene
+        .ccmd("md5sum")
+        .arg("--check")
+        .arg("--warn")
+        .arg(at.subdir.join("in.md5"))
+        .succeeds()
+        .stderr_contains("in.md5: 3: improperly formatted MD5 checksum line")
+        .stderr_contains("WARNING: 1 line is improperly formatted");
+
+    // with strict, we should fail the execution
+    scene
+        .ccmd("md5sum")
+        .arg("--check")
+        .arg("--strict")
+        .arg(at.subdir.join("in.md5"))
+        .fails();
+}
+
+#[test]
+fn test_check_status() {
+    let scene = TestScenario::new(util_name!());
+    let at = &scene.fixtures;
+
+    at.write("f", "");
+    at.write("in.md5", "MD5(f)= d41d8cd98f00b204e9800998ecf8427f\n");
+    scene
+        .ccmd("md5sum")
+        .arg("--check")
+        .arg("--status")
+        .arg(at.subdir.join("in.md5"))
+        .fails()
+        .stderr_is("")
+        .stdout_is("");
+}
+
+#[test]
+fn test_check_status_code() {
+    let scene = TestScenario::new(util_name!());
+    let at = &scene.fixtures;
+
+    at.write("f", "");
+    at.write("in.md5", "d41d8cd98f00b204e9800998ecf8427f  f\n");
+    scene
+        .ccmd("md5sum")
+        .arg("--check")
+        .arg("--status")
+        .arg(at.subdir.join("in.md5"))
+        .fails()
+        .stderr_is("")
+        .stdout_is("");
+}
+
+#[test]
+fn test_check_no_backslash_no_space() {
```

Comment:
I don't understand what this test is meant to test. The content of `in.md5` contains a space and while it's true that there is no backslash (except in `\n`) I don't see what the relevance is of there being no backslash.

Reply:
it mirrors one of the test in GNU